parse_test.go

```diff
@@ -0,0 +1,67 @@
+package goconf
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type Conf struct {
+	Name string `env:"MY_NAME"`
+	Age  int    `env:"MY_AGE"`
+	Team string `env:"MY_TEAM"`
+}
+
+func TestParseEnv(t *testing.T) {
+	updateEnv()
+	tests := []struct {
+		name           string
+		expectedOutput Conf
+		errorExpected  bool
+	}{
+		{
+			name: "Successfully parsed the configs",
+			expectedOutput: Conf{
+				Name: "coderx",
+				Age:  99,
+				Team: "backend",
+			},
+			errorExpected: false,
+		},
+		{
+			name:           "config parse failure scenario",
+			expectedOutput: Conf{},
+			errorExpected:  true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if !test.errorExpected {
+				conf := newConf()
+				err := ParseEnv(&conf)
+				assert.NoError(t, err)
+				assert.Equal(t, test.expectedOutput, conf)
+			} else {
+				confPointer := newPointerToConf()
+				err := ParseEnv(&confPointer)
+				assert.Error(t, err)
```

Comment:
I think we can simplyfy this part, also this test is not extendable if we write this manner. Also it's better to assert meaningful part of the error message to be sure that we are getting the expected error. WDYT getting the input from the test dataset?